src/ShadowDOM/wrappers/Document.js

```diff
@@ -338,6 +338,13 @@
     'hasFeature',
   ]);
 
+  var originalCreateTreeWalker = document.createTreeWalker;
+  function createTreeWalker(root, whatToShow, filter, entityReferenceExpansion){
+    return originalCreateTreeWalker.call(document,unwrap(root), whatToShow, filter, entityReferenceExpansion);
```

Comment:
one more thought: do we need to handle wrapping on the other arguments or the result here? In other words, if those APIs take Nodes, they need to deal with wrapping/unwrapping too.